pulser-simulation/pulser_simulation/simresults.py

```diff
@@ -525,11 +525,16 @@ def sample_state(
             return sampled_state
 
         detected_sample_dict: Counter = Counter()
+        eps = self._meas_errors["epsilon"]
+        eps_p = self._meas_errors["epsilon_prime"]
+
+        if eps_p == 0.0 and eps == 0.0:
+            return sampled_state
+
```

Comment:
I think you can just add this condition to the condition line 524:
`if self._meas_errors is None and eps_p == 0.0 and eps == 0.0:`

Reply:
@a-corni if _meas_errors is None, I don't have any eps_p / eps value; I could do something like this:

```python
if self._meas_errors is None or (self._meas_errors["epsilon"] == 0.0 and self._meas_errors["epsilon_prime"] == 0)
```

But IMO it is more readable as it is now

Reply:
But if `self._meas_errors is None` there is no `eps` and  `eps_p` though

Reply:
To avoid the duplicated return, I think it needs to be something like this:
```python
if self._meas_errors is None or (self._meas_errors["epsilon_p"] == 0.0 and self._meas_errors["epsilon"] == 0.0):
```

Reply:
Do you mean like this? If you confirm I proceed with this edit.

```python
if self._meas_errors is None or (self._meas_errors["epsilon"] == 0.0 and self._meas_errors["epsilon_prime"] == 0)
```

@a-corni I requested HG review because was the only name showing in the request review list, feel free to take the task.

Reply:
Yes that's what I meant ;)

Reply:
Don't forget about this :)